### ISSUE TITLE
Architecture documentation

### ISSUE THREAD

**Issue by wtrocki** [morrys/wora#2]
Can you provide more info how you see offline functionalities being developed and how you see future of this library going forward? 
Any info will be useful?

**Comment by morrys** [comment]
Soon I will create a roadmap within github to highlight the macro themes that will be addressed by the individual libraries.

The architecture and related documentation will be extended and integrated according to the needs of the community.

As for the apollo libraries, the features introduced extend the original ones without altering the behavior of the client when the network status is online.

**Comment by wtrocki** [comment]
Yes. I like the idea. Any reason why users will need to use custom replication protocol in execute? I see that you can change this to use the same graphql mutations and url.

**Comment by wtrocki** [comment]
> Soon I will create a roadmap within github to highlight the macro themes that will be addressed by the individual libraries.

This sounds amazing!

**Comment by morrys** [comment]
are you talking about **link: httpLinkOffline** in https://github.com/morrys/wora/tree/master/packages/apollo-offline#apollo-client-with-offline-options ?

**Comment by wtrocki** [comment]
Yes.

**Comment by morrys** [comment]
it is not mandatory to enter this option, by default the same link is used to execute mutations online.

I gave this opportunity just to give the possibility to insert custom logics for offline mutations.
For example, different retry links, different error handling, payload check or all those needs that may arise based on the use case.

**Comment by morrys** [comment]
I could tell you a thousand possible solutions of use but without a requirement and / or a problem it is useless.

**Comment by wtrocki** [comment]
That is a really interesting approach. Is this global handler then or we can apply this with UI binding. I did not mean this so sound like bad idea - I want to see what are the boundaries and user responsibilities and map the problems I know to the wora.

**Comment by wtrocki** [comment]
Checked example: https://github.com/morrys/offline-examples/blob/master/apollo/todo/client/src/components/TodoOffline.js

It looks like there will be couple migration steps require to use it starting from vanilla Apollo application. It will be easy to avoid that IMHO assuming that offline mutations are not blocking.

**Comment by morrys** [comment]
Queuing occurs asynchronously and responds to the UI with the optimistic response. It does not keep the ui locked and absolutely does not wait for any response from the server.

Execution is performed in the background once the connection becomes available (and it is also possible to manage it manually).

**Comment by morrys** [comment]
https://github.com/morrys/wora/blob/master/packages/apollo-offline/src/ApolloStoreOffline.ts#L155

**Comment by morrys** [comment]
`Checked example: https://github.com/morrys/offline-examples/blob/master/apollo/todo/client/src/components/TodoOffline.js`

this react component is the one I talk about in this comment: https://github.com/morrys/wora/issues/3#issuecomment-512357653

**Comment by morrys** [comment]
to use the library this class is enough:

https://github.com/morrys/offline-examples/blob/master/apollo/todo/client/src/apollo/index.js

and this:

https://github.com/morrys/offline-examples/blob/master/apollo/todo/client/src/app.js#L65-L68
